Examples/KleinGordon/KleinGordonLevel.cpp

```diff
@@ -0,0 +1,186 @@
+#include "KleinGordonLevel.hpp"
+#include "FixedGridsTagger.hpp"
+#include "FourthOrderDerivatives.hpp"
+#include "KleinGordonRHS.hpp"
+#include <numeric>
+
+using namespace amrex;
+
+void KleinGordonLevel::variableSetUp()
+{
+    BL_PROFILE("KleinGordonLevel::variableSetUp()");
+
+    // Set up the state variables
+    stateVariableSetUp();
+
+    // The first two derived variables calculate the analytic solution
+    //  for phi and Pi
+
+    const std::string &comp_type                = {"analytic_soln"};
+    const amrex::Vector<std::string> comp_names = {"phi_analytic",
+                                                   "Pi_analytic"};
+
+    int ncomp_analytic{
+        static_cast<int>(comp_names.size())}; // how many derived variables
+
+    derive_lst.add(
+        comp_type, amrex::IndexType::TheCellType(), ncomp_analytic, comp_names,
+        calc_analytic_solution, [=](const amrex::Box &box)
+        { return amrex::grow(box, simParams().num_ghosts); },
+        &amrex::cell_quartic_interp);
+    derive_lst.addComponent("analytic_soln", desc_lst, State_Type, 0, 1);
+
+    // The following is an example of how to use the current state to compute a
+    // new derived variable that depends on the state variables and the
+    // potential
+
+    amrex::ParmParse pp;
+    std::string model{};
+    pp.query("model", model);
+
+    const int ncomp_rho{1}; // only one component associated with energy density
+
+    if (model == "Wave")
+    {
+        derive_lst.add(
+            "rho", amrex::IndexType::TheCellType(), 1,
```

Comment:
Might as well use the variable defined above rather than a magic number.
```suggestion
            "rho", amrex::IndexType::TheCellType(), ncomp_rho,
```